drivers/ethernet/eth_stm32_hal.c

```diff
@@ -773,6 +775,103 @@ static int eth_initialize(const struct device *dev)
 	return 0;
 }
 
+static uint32_t reverse(uint32_t val)
+{
+	uint32_t res = 0;
+	int i;
+
+	for (i = 0; i < 32; i++) {
+		if (val & (1 << i)) {
+			res |= 1 << (31 - i);
+		}
+	}
+
+	return res;
+}
+
+static void net_if_mcast_cb(struct net_if *iface,
+			    const struct net_addr *addr,
+			    bool is_joined)
+{
+	ARG_UNUSED(addr);
+	ARG_UNUSED(is_joined);
+
+	const struct device *dev;
+	struct eth_stm32_hal_dev_data *dev_data;
+	ETH_HandleTypeDef *heth;
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if_ipv6 *ipv6;
+#endif
+#if defined(CONFIG_NET_NATIVE_IPV4)
+	struct net_if_ipv4 *ipv4;
+#endif
+	struct net_eth_addr mac_addr;
+	uint32_t crc;
+	uint32_t hash_table[2];
+	uint32_t hash_index;
+	int i;
```

Comment:
```suggestion
	uint32_t i;
```
The used ´i´ variable is always positive here!